From 599d471e1119f3606739749f151070a061c831a7 Mon Sep 17 00:00:00 2001
From: John Hood <cgull@glup.org>
Date: Tue, 9 Jul 2019 00:30:16 -0400
Subject: [PATCH] Move generated includes to their own directory to avoid
 conflicts.

Fixes #1051, mosh fails to build on case-insensitive filesystems.

XXX This isn't perfect because autoconf/automake drop several extra
files in the directory anyway.
---
 Makefile.am             | 13 +++----------
 configure.ac            |  3 ++-
 src/Makefile.am         |  2 +-
 src/include/Makefile.am | 14 ++++++++++++++
 4 files changed, 20 insertions(+), 12 deletions(-)
 create mode 100644 src/include/Makefile.am

diff --git Makefile.am Makefile.am
index ac0bb8f..52726c6 100644
--- Makefile.am
+++ Makefile.am
@@ -1,7 +1,7 @@
 ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = scripts src man conf
 EXTRA_DIST = autogen.sh ocb-license.html README.md COPYING.iOS
-BUILT_SOURCES = version.h
+BUILT_SOURCES = VERSION
 CLANG_SCAN_BUILD = scan-build
 AM_DISTCHECK_CONFIGURE_FLAGS = --enable-compile-warnings=distcheck --enable-examples
 
@@ -19,18 +19,11 @@ VERSION:
 	fi
 	@rm -f VERSION.dist VERSION.git
 
-version.h:	VERSION
-	@printf '#define BUILD_VERSION "%s"\n' "$$(cat VERSION)" > version.h.new
-	@set -e; if ! diff -q version.h version.h.new > /dev/null 2>&1; then \
-		mv -f version.h.new version.h; \
-	fi
-	@rm -f version.h.new
-
 clean-local:
-	@rm -rf version.h VERSION cov-int mosh-coverity.txz
+	@rm -rf VERSION cov-int mosh-coverity.txz
 
 cppcheck: $(BUILT_SOURCES) config.h
-	cppcheck --enable=all --template=gcc -include config.h -I . \
+	cppcheck --enable=all --template=gcc -include src/include/config.h -I src/include \
 		-I src/crypto -I src/frontend -I src/network -I src/protobufs \
 		-I src/statesync -I src/terminal -I src/util \
 		-I /usr/include -I /usr/include/google/protobuf -I/usr/include/openssl \
diff --git configure.ac configure.ac
index 3ad983d..0304e5f 100644
--- configure.ac
+++ configure.ac
@@ -7,7 +7,7 @@ AM_INIT_AUTOMAKE([foreign std-options -Wall -Werror])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_CONFIG_SRCDIR([src/frontend/mosh-client.cc])
 AC_CONFIG_MACRO_DIR([m4])
-AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_HEADERS([src/include/config.h])
 
 # Checks for programs.
 AC_PROG_CC([cc gcc clang])
@@ -490,6 +490,7 @@ AC_CONFIG_FILES([
   src/Makefile
   src/crypto/Makefile
   src/frontend/Makefile
+  src/include/Makefile
   src/network/Makefile
   src/protobufs/Makefile
   src/statesync/Makefile
diff --git src/Makefile.am src/Makefile.am
index 2390f7c..a714997 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -1 +1 @@
-SUBDIRS = protobufs util crypto terminal network statesync frontend examples tests
+SUBDIRS = include protobufs util crypto terminal network statesync frontend examples tests
diff --git src/include/Makefile.am src/include/Makefile.am
new file mode 100644
index 0000000..98d140f
--- /dev/null
+++ src/include/Makefile.am
@@ -0,0 +1,14 @@
+BUILT_SOURCES = version.h
+nodist_INCLUDES = version.h
+
+clean-local:
+	@rm -rf version.h
+
+version.h:	../../VERSION
+	@test -f $<
+	@printf '#define BUILD_VERSION "%s"\n' "$$(cat $<)" > version.h.new
+	@set -e; if ! diff -q $@ $@.new > /dev/null 2>&1; then \
+		mv -f $@.new $@; \
+	fi
+	@rm -f $@.new
+
-- 
2.20.1 (Apple Git-117)

